### ISSUE TITLE
Rename parent folder following maven convention

### ISSUE THREAD

**Issue by glalloue** [cnumr/ecoCode#74]
Hi,

I'm look since few weeks the project and i have identified a recent renaming that i'm not agreed. I'm hoping my remark will be accepted/appreciated :)

You are using maven structure. The parent folder **src** is not appropriate. By convention, we're expected a pom together with a **src** folder. 
And inside the pom, we are expected a **ProjectName** or **Artifactid** equal to the name folder (WYSIWYG).
More info and schema expected : [wikipedia Maven](https://en.wikipedia.org/wiki/Apache_Maven)

It will be more efficiency to rename as "ecocode" the **src** parent folder or any suggestion idea you have in mind. :)

_Originally posted by @Silicoman in https://github.com/cnumr/ecoCode/discussions/72_

**Comment by cyChop** [comment]
This should apply to modules as well. Right now, the folder names and artifactIds of modules are not aligned.

**Comment by dedece35** [comment]
hello,
I'm agree with observation and I have been surprised to see this directory structure and pom.xml content.
"ecocode" name is already used by the git project. Thus, I don't think this is a good idea to rename "src" directory to "ecocode" ( because we will have a "ecocode/ecocode" structure :( )

I think, we have to rethink the directpry structure but if we want smallest possibles impacts, indeed, the simplest way is to rename "src" directory  : I suggest "dev" or "src-code" or anything else that suggests this directory contains source code.

Longer term, I think source code must be in root directory (as mentionned in maven guidelines) and some others directories for documentation, installation files, deploy files (for example : final plugin JARs should be in a "target" directory instead of "lib" directory if we want to follow maven guidelines).
We have to discuss about it.

**Comment by dedece35** [comment]
an idea of code architecture 
![ecocode-archi drawio](https://user-images.githubusercontent.com/12011401/200679300-ea60bf71-5fc8-480b-b450-b58d2a6f86d5.png)

**Comment by jhertout** [comment]
I just want to indicate that the plugin-android scans 3 languages: java, xml, groovy and will support kotlin in the futur. I am not sure the "language" entry point is the best. The same is true for "plugin-ios" which could have several language in it.

If we abslolutly want to reorganize the repo, for the moment I think that "plugin-ios" and "plugin-android" should be at the same level as the web languages plugins "plugin-php"...

An other point is that in the "android-plugin" we do not work at the "language" level like in the "plugin-java" for example but at the API level (we are looking directly for bad usage of methods in the Android Bluetooth API for example).
I think that it may happen in the web plugins if for example we decide to work on a "plugin-django" or a "plugin-reactjs" (It is just for the example :) ). The approach is different from the classical "language" approach of SonarQube but I think that it is necessary since (in Android at least) the true eco-conception problems came from the API.

**Comment by dedece35** [comment]
@MP-Aubay @jhertout 
My draw is just for initiate the discussion but I'm Ok that we must have a live internal discussion to talk about it.

@glalloue could you organize it and see who wants to participate to this live meeting ?

**Comment by jhertout** [comment]
@dedece35 
I agree that we must reorganize the architecture of the project. I just write the reflections we have when working on Android. I am not sure it can be applied to web plugins too ;)

I agree that a live discussion will be more efficient on that subject.

**Comment by dedece35** [comment]
discussion in progress in core team
